File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	StringTokenizer st = new StringTokenizer(br.readLine());
    	int N= Integer.parseInt(st.nextToken());
    	int M= Integer.parseInt(st.nextToken());
    	int arr [][] = new int[N][M];
    	for(int i=0; i<N; i++) {
    		st = new StringTokenizer(br.readLine());
    		for(int j=0; j<M; j++) {
    			arr[i][j] = Integer.parseInt(st.nextToken());
    		}
    	}
    	int K = Integer.parseInt(br.readLine());
    	for(int a=0; a<K; a++) {
    		st = new StringTokenizer(br.readLine());
    		int i=Integer.parseInt(st.nextToken())-1;
    		int j=Integer.parseInt(st.nextToken())-1;
    		int x=Integer.parseInt(st.nextToken())-1;
    		int y=Integer.parseInt(st.nextToken())-1;
    		int sum = 0;
    		for (int p = i; p <= x; p++) {
    			for (int q = j; q <= y; q++) {
    				sum += arr[p][q];
    			}
    		}
    		
    		System.out.println(sum);
    	}
    }
}

File: 백준/Silver/2167. 2차원 배열의 합/2차원 배열의 합.c
#include <stdio.h>

int main() {
    int N, M;
    scanf("%d %d", &N, &M);
    
    int arr[N][M];
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < M; j++) {
            scanf("%d", &arr[i][j]);
        }
    }
    
    int K;
    scanf("%d", &K);
    
    for (int a = 0; a < K; a++) {
        int i, j, x, y;
        scanf("%d %d %d %d", &i, &j, &x, &y);
        i--; j--; x--; y--;
        int sum = 0;
        for (int p = i; p <= x; p++) {
            for (int q = j; q <= y; q++) {
                sum += arr[p][q];
            }
        }
        printf("%d\n", sum);
    }
    
    return 0;
}
